src/test/java/com/newrelic/telemetry/opentelemetry/export/MetricPointAdapterTest.java

```diff
@@ -0,0 +1,174 @@
+            Descriptor.create(
+                "metricName",
+                "metricDescription",
+                "units",
+                Type.MONOTONIC_DOUBLE,
+                singletonMap("commonKey", "commonValue")),
+            Type.MONOTONIC_DOUBLE,
+            commonAttributes,
+            DoublePoint.create(
+                100,
+                TimeUnit.MILLISECONDS.toNanos(10_000L),
+                singletonMap("specificKey", "specificValue"),
+                123.55d));
+
+    Attributes expectedAttributes =
+        new Attributes().put("service.name", "fooService").put("specificKey", "specificValue");
+    assertEquals(
+        singleton(new Count("metricName", 123.55d, 9_000L, 10_000L, expectedAttributes)), result);
+  }
+
+  @Test
+  void testDoublePoint_nonMonotonic() throws Exception {
+    TimeTracker timeTracker = mock(TimeTracker.class);
+    MetricPointAdapter metricPointAdapter = new MetricPointAdapter(timeTracker);
+
+    Attributes commonAttributes = new Attributes().put("service.name", "fooService");
+    Collection<Metric> result =
+        metricPointAdapter.buildMetricsFromPoint(
+            Descriptor.create(
+                "metricName",
+                "metricDescription",
+                "units",
+                Type.NON_MONOTONIC_DOUBLE,
+                singletonMap("commonKey", "commonValue")),
+            Type.NON_MONOTONIC_DOUBLE,
+            commonAttributes,
+            DoublePoint.create(
+                100,
+                TimeUnit.MILLISECONDS.toNanos(10_000L),
+                singletonMap("specificKey", "specificValue"),
+                123.55d));
+
+    Attributes expectedAttributes =
+        new Attributes().put("service.name", "fooService").put("specificKey", "specificValue");
+    assertEquals(singleton(new Gauge("metricName", 123.55d, 10_000L, expectedAttributes)), result);
+  }
+
+  @Test
+  void testSummaryPoint() throws Exception {
+    TimeTracker timeTracker = mock(TimeTracker.class);
+    when(timeTracker.getPreviousTime()).thenReturn(TimeUnit.MILLISECONDS.toNanos(9_000L));
+    MetricPointAdapter metricPointAdapter = new MetricPointAdapter(timeTracker);
+
+    Attributes commonAttributes = new Attributes().put("service.name", "fooService");
+    Collection<Metric> result =
+        metricPointAdapter.buildMetricsFromPoint(
+            Descriptor.create(
+                "metricName",
+                "metricDescription",
+                "units",
+                Type.SUMMARY,
+                singletonMap("commonKey", "commonValue")),
+            Type.SUMMARY,
+            commonAttributes,
+            SummaryPoint.create(
+                100,
+                TimeUnit.MILLISECONDS.toNanos(10_000L),
+                singletonMap("specificKey", "specificValue"),
+                200,
+                123.55d,
+                Arrays.asList(
+                    ValueAtPercentile.create(0.0, 5.5d),
+                    ValueAtPercentile.create(100.0, 100.01d))));
```

Comment:
I think a few local vars could help the readability here...it's a lot of data!

Reply:
agreed. Makes it easier to see the structure if I do, as well.

Reply:
fixed. please take a look

Reply:
👍